src/hdmf/backends/hdf5/h5tools.py

```diff
@@ -134,13 +145,16 @@ def __check_path_file_obj(cls, path, file_obj):
         return path
 
     @classmethod
-    def __resolve_file_obj(cls, path, file_obj, driver):
+    def __resolve_file_obj(cls, path, file_obj, driver, aws_region=None):
```

Comment:
Codecov complaining of lack of hit on this line - unsure how to add a test for that

So I'll let @rly, @oruebel , or @mavaylon1 add that or accept with reduced coverage

Reply:
The ``__resolve_file_obj`` function seems to be used only by the `load_namespaces` and `get_namespaces` function. So to test the function for `aws` you could add a test for `ros3` that calls one of those functions. However, I think the simpler approach may be to just have a test that calls the method on its own and asserts that the file is opened correctly; something like"

```
fileobj = HDF5IO._HDF5IO__resolve_file_obj(path=<s3path>, file_obj=None, driver='ros3', aws_region=<you aws region>)

assertIsInstance(fileobj, h5py.File)
assertEquals(fileobj.path, <s3path>)
```

Reply:
I just updated `load_namespaces` and `get_namespaces` to take `aws_region` and pass it to `__resolve_file_obj`. Otherwise an s3 path will not open.

Reply:
I think all of our tests use http instead of S3 paths, so this is new. I'll add some tests today/this weekend.

Reply:
@rly while you are at it, could you also add a docstring for `__resolve_file_obj`. It is currently not clear that this function is specific for the loading of namespaces and it would be helpful to have that documented.